Frameworks/Foundation/NSString.mm

```diff
@@ -1093,7 +1097,15 @@ - (NSRange)rangeOfString:(NSString*)searchString options:(NSStringCompareOptions
  @Status Interoperable
 */
 - (BOOL)isAbsolutePath {
-    return ([self hasPrefix:_NSGetSlashStr()]) || ((_isLetter([self characterAtIndex:0])) && ([self characterAtIndex:1] == ':'));
+    if ([self hasPrefix:_NSGetSlashStr()]) {
+        return YES;
+    }
+
+    if (self.length >= 2) {
+        return _isLetter([self characterAtIndex:0]) && [self characterAtIndex:1] == ':';
```

Comment:
Talked to DHowett offline, he's going to take a closer look at drive paths next iteration, and leave this slightly-wrong state for now.

Be sure to add a comment about this status.